asQ/paradiag.py

```diff
@@ -192,5 +192,7 @@ def solve(self,
             if wndw != nwindows-1:
                 self.aaofunc.bcast_field(-1, self.aaofunc.initial_condition)
                 self.aaofunc.assign(self.aaofunc.initial_condition)
-
+                for n in range(self.aaofunc.nlocal_timesteps):
```

Comment:
Because we specify the `dt` in the form `t0` and `time` are not independent, so they must always be updated at the same time to be consistent. I think this means that updating the t0 and time values warrants it's own method on the Form. Something like `aaoform.update_initial_time(t)` that updates `t0` and the `time` array accordingly.

Users won't always be using the Paradiag class and this method would also reduce the burden (and likelihood of typos) on them.